pmd-java/src/main/java/net/sourceforge/pmd/lang/java/metrics/api/JavaClassMetricKey.java

```diff
@@ -101,7 +101,18 @@ public String name() {
             public boolean supports(ASTAnyTypeDeclaration node) {
                 return metric.supports(node);
             }
+
+
+            @Override
+            public boolean equals(Object obj) {
+                return obj == this;
```

Comment:
Yes you're right
I think I intended to trust the programmer with managing their instances but it's better your way